File: scripts/swift/reduce.swift
func doubleMapReduce(_ numbers: [Int]) {
    print("\nFunctional style map/reduce:")
    let array = numbers.map() { number in
        number * 2
    }
    print("Doubles: \(array)")
    let sum = array.reduce(0, { x, y in
        x + y
    })
    print("Sum of doubles: \(sum)")
}

func doubleMapReduceChain(_ numbers: [Int]) {
    print("\nFunctional style map/reduce chaining:")
    let sum = numbers.map() { number in
        number * 2
    }.reduce(0, { x, y in
        x + y
    })
    print("Sum of doubles: \(sum)")
}

//
// ----- Start of Script -----
//

let numbers = [1, 2, 3, 4, 5]
print("Numbers: \(numbers)")

doubleMapReduce(numbers)
doubleMapReduceChain(numbers)

File: FunctionalStyle/javascript/reduce.js
function doubledMapReduce(numbers) {
  console.log("\nFunctional style map/reduce:");
  const array = numbers.map((n) => {
    return n * 2;
  });
  console.log("Doubles:", array);
  const sum = array.reduce(
    (accumulator, currentValue) => accumulator + currentValue,
    0
  );
  console.log("Sum of double values:", sum);
}

function doubledMapReduceChain(numbers) {
  console.log("\nFunctional style map/reduce chaining:");
  const sum = numbers
    .map((n) => {
      return n * 2;
    })
    .reduce((accumulator, currentValue) => accumulator + currentValue, 0);
  console.log("Sum of double values:", sum);
}

//
// ----- Start of Script -----
//

const numbers = [1, 2, 3, 4, 5];
console.log("Numbers:", numbers);

doubledMapReduce(numbers);
doubledMapReduceChain(numbers);
